test/weights.jl

```diff
@@ -3,355 +3,376 @@ using Base.Test
 using Compat
 import Compat: view
 
-@test isa(weights([1, 2, 3]), WeightVec{Int})
-@test isa(weights([1., 2., 3.]), WeightVec{Float64})
-@test isa(weights([1 2 3; 4 5 6]), WeightVec{Int})
-
-@test isa(WeightVec([1, 2, 3], 6), WeightVec{Int})
-
-@test isempty(weights(Float64[]))
-@test size(weights([1, 2, 3])) == (3,)
-
-w  = [1., 2., 3.]
-wv = weights(w)
-@test eltype(wv) === Float64
-@test length(wv) === 3
-@test values(wv) === w
-@test sum(wv) === 6.0
-@test !isempty(wv)
-
-b  = trues(3)
-bv = weights(b)
-@test eltype(bv) === Bool
-@test length(bv) === 3
-@test values(bv) === b
-@test sum(bv)    === 3
-@test !isempty(bv)
-
-ba = BitArray([true, false, true])
-sa = sparsevec([1., 0., 2.])
-
-@test sum(ba, wv) === 4.0
-@test sum(sa, wv) === 7.0
-
-## wsum
-
-x = [6., 8., 9.]
-w = [2., 3., 4.]
-p = [1. 2. ; 3. 4.]
-q = [1., 2., 3., 4.]
-
-@test wsum(Float64[], Float64[]) === 0.0
-@test wsum(x, w) === 72.0
-@test wsum(p, q) === 29.0
+@testset "StatsBase.Weights" begin
+weight_funcs = (aweights, fweights, pweights)
+
+@testset "Construction" begin
+    @testset "$f" for f in weight_funcs
+        @test isa(f([1, 2, 3]), AbstractWeights{Int})
+        @test isa(f([1., 2., 3.]), AbstractWeights{Float64})
+        @test isa(f([1 2 3; 4 5 6]), AbstractWeights{Int})
+
+        @test isempty(f(Float64[]))
+        @test size(f([1, 2, 3])) == (3,)
+
+        w  = [1., 2., 3.]
+        wv = f(w)
+        @test eltype(wv) === Float64
+        @test length(wv) === 3
+        @test values(wv) === w
+        @test sum(wv) === 6.0
+        @test !isempty(wv)
+
+        b  = trues(3)
+        bv = f(b)
+        @test eltype(bv) === Bool
+        @test length(bv) === 3
+        @test values(bv) === b
+        @test sum(bv)    === 3
+        @test !isempty(bv)
+
+        ba = BitArray([true, false, true])
+        sa = sparsevec([1., 0., 2.])
+
+        @test sum(ba, wv) === 4.0
+        @test sum(sa, wv) === 7.0
+    end
 
-## wsum along dimensions
+    @testset "eweights" begin
+        λ = 0.2
+        wv = eweights(4, λ)
+        @test round(values(wv), 4) == [0.2, 0.25, 0.3125, 0.3906]
```

Comment:
Copy the full precision numbers rather than rounding. That could help spotting precision issues if somebody tweaks the formula.

Should also check the type of `wv`.